$NetBSD: patch-CMakeLists.txt,v 1.2 2026/09/10 08:36:13 markd Exp $

Configure like Linux for some defaults

Add Qt6 Core5Compat as needed but not pulled in if using external quazip1

--- CMakeLists.txt.orig	2026-08-14 17:43:46.000000000 +0000
+++ CMakeLists.txt
@@ -44,6 +44,8 @@ endif()
 include(cmake/project_version.cmake)
 include(GNUInstallDirs)
 
+set(ADDITIONAL_DEFINITIONS ${ADDITIONAL_DEFINITIONS} -DQ_WS_X11)
+
 if(MSVC)
 	set(ADDITIONAL_DEFINITIONS ${ADDITIONAL_DEFINITIONS} -D_CRT_SECURE_NO_WARNINGS -D_UNICODE -DUNICODE)
 	add_compile_options(/wd4267 /wd4244)
@@ -66,6 +68,7 @@ find_package(Qt${QT_VERSION_MAJOR} COMPO
 find_package(Qt${QT_VERSION_MAJOR} COMPONENTS PrintSupport Svg Qml REQUIRED)
 if(${QT_VERSION_MAJOR} GREATER_EQUAL 6)
     find_package(Qt${QT_VERSION_MAJOR} COMPONENTS LinguistTools QUIET)
+    find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Core5Compat QUIET)
 endif()
 
 message(STATUS "building for Qt ${QT_VERSION_MAJOR}")
@@ -394,6 +397,7 @@ target_link_libraries(texstudio PRIVATE
     Qt${QT_VERSION_MAJOR}::Qml
     Qt${QT_VERSION_MAJOR}::Concurrent
     Qt${QT_VERSION_MAJOR}::UiTools
+    Qt${QT_VERSION_MAJOR}::Core5Compat
 )
 if(NOT MSVC AND CMAKE_VERSION VERSION_GREATER_EQUAL "3.16.0")
     target_precompile_headers(texstudio PUBLIC src/mostQtHeaders.h)
